### PR TITLE
Provision_behavior_provisioningdata-Expiration_policies-Role_status

### PR DESCRIPTION
**Pull request by ioigoume**
- Calculate change diff and pass in provisioning data
- CoPersonRole Status Fix
- Expiration Policy Changelog Behavior
```sql
alter table cm_co_expiration_policies
    add co_expiration_policy_id int;
alter table cm_co_expiration_policies
    add deleted boolean default false;
alter table cm_co_expiration_policies
    add actor_identifier varchar(256);
alter table cm_co_expiration_policies
    add revision int default 0;
create index cm_co_expiration_policies_i2
    on cm_co_expiration_policies (co_expiration_policy_id);
alter table cm_co_expiration_policies
    add constraint cm_co_expiration_policies_co_expiration_policy_id_fkey
        foreign key (co_expiration_policy_id) references cm_co_expiration_policies;
```

### CODE REVIEW

File: app/Model/Behavior/ProvisionerBehavior.php [hunk 1]
```diff
@@ -1129,7 +1154,7 @@ protected function provisionEmailLists($model, $coEmailListIds, $created, $provi
    * @throws InvalidArgumentException
    */
   
-  protected function provisionGroups($model, $gmodel, $coGroupIds, $created, $provisioningAction, $coPersonId=null) {
+  protected function provisionGroups($model, $gmodel, $coGroupIds, $created, $provisioningAction, $coPersonId=null, $originChange=array()) {
```
Comment (lionick) [inline app/Model/Behavior/ProvisionerBehavior.php:1159]: update phpDoc with the new argument, and maybe the description of the function if required

File: app/Model/Behavior/ProvisionerBehavior.php [hunk 2]
```diff
@@ -1185,7 +1211,7 @@ protected function provisionGroups($model, $gmodel, $coGroupIds, $created, $prov
    * @throws InvalidArgumentException
    */
   
-  protected function provisionPeople($model, $pmodel, $coPersonIds, $created, $provisioningAction) {
+  protected function provisionPeople($model, $pmodel, $coPersonIds, $created, $provisioningAction, $originChange=array()) {
```
Comment (lionick) [inline app/Model/Behavior/ProvisionerBehavior.php:1218]: update the phpDoc and maybe the description of the function

**Review by lionick — CHANGES_REQUESTED** [review]

File: app/Model/Behavior/ProvisionerBehavior.php (hunk 2 above)
Comment (ioigoume) [reply app/Model/Behavior/ProvisionerBehavior.php:1218]: Done

File: app/Model/Behavior/ProvisionerBehavior.php (hunk 1 above)
Comment (ioigoume) [reply app/Model/Behavior/ProvisionerBehavior.php:1159]: Done

**Review by lionick — APPROVED** [review]
LGTM

**Review by NicolasLiampotis — APPROVED** [review]
LGTM

**Pull request merged**